### PR TITLE
Make IPython work with OpenSSL in FIPS mode

### LINKED ISSUE #10615: ipython does not work with python compiled with openssl in fips mode.
in fips mode, md5 is disabled. hence, ipython fails with 

```
    hash_digest = hashlib.md5(code.encode("utf-8")).hexdigest()
ValueError: error:060800A3:digital envelope routines:EVP_DigestInit_ex:disabled for fips
```
this error.

A fix is changing md5 to sha256 in `IPython/core/compilerop.py` [in this line](https://github.com/ipython/ipython/blob/master/IPython/core/compilerop.py#L58). I can raise a pull request if the change looks fine.

### PR DESCRIPTION
**Pull request by srinathgs**
`md5` is not supported with OpenSSL in FIPS mode, hence moving `md5` to `sha1`

closes  #10615

### CODE REVIEW

**Comment by Carreau** [comment]
+1, I'll let it sit a bit just to get review and marked it as for backport. As soon as I'm back from holidaysI'll try to merge a couple of PR and make minor releases of IPython 5.x and 6.x

Thanks .

**Review by ivanov — APPROVED** [review]
Thanks for getting back to this @srinathgs, glad to have your first contribution in IPython! Woo!

**Comment by takluyver** [comment]
@meeseeksdev backport

**Pull request merged**
